src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,43 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<? extends Machine> machines;
+        switch (type.getSimpleName()) {
+            case "Bulldozer":
+                machines = new BulldozerProducer().get();
```

Comment:
```suggestion
                return new ArrayList<>(bulldozerProducer().get());
```
better to make producer a class variable not to create new instance of producer per each method call